test/test_Path.py

```diff
@@ -41,48 +42,51 @@ def createState(spaceInformation, xy):
-        reachedPos = utils.XYToLatlon(xy=(4.5, 4.5), referenceLatlon=path.getReferenceLatlon())
+        reachedPos = utils.XYToLatlon(xy=(boundaryCoord + 0.01, boundaryCoord + 0.01), referenceLatlon=path.getReferenceLatlon())
         self.assertTrue(path.nextWaypointReached(reachedPos))
         self.assertTrue(path.lastWaypointReached(reachedPos))
 
         # Test notReachedPos
-        notReachedPos = utils.XYToLatlon(xy=(4, 4), referenceLatlon=path.getReferenceLatlon())
+        notReachedPos = utils.XYToLatlon(xy=(boundaryCoord - 0.01, boundaryCoord - 0.01), referenceLatlon=path.getReferenceLatlon())
         self.assertFalse(path.nextWaypointReached(notReachedPos))
         self.assertFalse(path.lastWaypointReached(notReachedPos))
 
     # testing cases where start and LWP have same lat or same lon
     def test_nextWaypointReached_sameLat(self):
-        # Setup path from (1,1) to (1,3)
-        # Test passes when distance to next waypoint is less than or equal to 1
-        path = self.createPathWithGivenPositionsXY([[1, 1], [1, 3]])
+        # Setup path from (1,1) to (1,100)
+        # Test passes when distance to next waypoint is less than or equal to WAYPOINT_REACHED_DISTANCE
+        path = self.createPathWithGivenPositionsXY([[1, 1], [1, 100]])
+        boundaryCoord = 100 - WAYPOINT_REACHED_DISTANCE
 
         # Test reachedPos
-        reachedPos = utils.XYToLatlon(xy=(100, 2.1), referenceLatlon=path.getReferenceLatlon())
+        reachedPos = utils.XYToLatlon(xy=(100, boundaryCoord + 0.01), referenceLatlon=path.getReferenceLatlon())
         self.assertTrue(path.nextWaypointReached(reachedPos))
         self.assertTrue(path.lastWaypointReached(reachedPos))
 
         # Test notReachedPos
-        notReachedPos = utils.XYToLatlon(xy=(-100, 1.9), referenceLatlon=path.getReferenceLatlon())
+        notReachedPos = utils.XYToLatlon(xy=(-100, boundaryCoord - 0.01), referenceLatlon=path.getReferenceLatlon())
         self.assertFalse(path.nextWaypointReached(notReachedPos))
         self.assertFalse(path.lastWaypointReached(notReachedPos))
 
     def test_nextWaypointReached_sameLon(self):
-        # Setup path from (3,1) to (1,1)
-        # Test passes when distance to next waypoint is less than or equal to 1
-        path = self.createPathWithGivenPositionsXY([[3, 1], [1, 1]])
+        # Setup path from (100,1) to (1,1)
+        # Test passes when distance to next waypoint is less than or equal to WAYPOINT_REACHED_DISTANCE
+        path = self.createPathWithGivenPositionsXY([[100, 1], [1, 1]])
+        boundaryCoord = 1 + WAYPOINT_REACHED_DISTANCE
 
         # Test reachedPos
-        reachedPos = utils.XYToLatlon(xy=(0.9, -100), referenceLatlon=path.getReferenceLatlon())
+        reachedPos = utils.XYToLatlon(xy=(boundaryCoord - 0.01, -100), referenceLatlon=path.getReferenceLatlon())
         self.assertTrue(path.nextWaypointReached(reachedPos))
         self.assertTrue(path.lastWaypointReached(reachedPos))
```

Comment:
I am having some trouble with this test. It passes when `boundaryCoord == 1`, but it should take into account `WAYPOINT_REACHED_DISTANCE` as well. Is there a way where I can see the values of the variables in path? I tried printing when running `catkin_make run_tests`, but that didn't work.